### ISSUE TITLE
Forbid `yield` keyword in `__init__` method

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#254]
# Rule request

## Thesis

It is currently possible to write:

```python
class ModuleMembersVisitor(object):

    def __init__(self, *args, **kwargs):
        self._public_items_count = 0
        yield 10
```

Both `flake8` and `mypy` will ignore this.

## Reasoning

Our initializer clearly should not be a generator. Since it breaks a lot of OOP patterns.

**Comment by a96tudor** [comment]
Hey! I'd like to work on this.

**Comment by sobolevn** [comment]
@a96tudor awesome!

**Comment by a96tudor** [comment]
I assume this would be considered as a best-practice violation, right?

**Comment by sobolevn** [comment]
Yes, I think so too.

**Comment by Shikhov-P** [comment]
@sobolevn  I've solved this issue https://github.com/wemake-services/wemake-python-styleguide/pull/279
